### ISSUE TITLE
Traing strategy

### ISSUE THREAD

**Issue by LaiSongxuan** [masyagin1998/robin#3]
The pretrained weight works quit well for various document samples. However, when I train my own network from scratch, it seems that the performance is always worse than the provided weight. Therefore, can you please tell me the optimization method and training strategy you used?

**Comment by masyagin1998** [comment]
Hi!
If you want same efficiency as on my weights, you should use data augmentation (there is command line arguments flag in training script for enabling augmentation with many features (main of them is double-inversion augmentation), and use mix of datasets: all DIBCO datasets, part of LRDE dataset (really small part) and other images.

**Comment by deepseek** [comment]
- Do you mean `--augmentate` & `--extraprocesses`?
- What does `--extraprocesses` do?

**Comment by masyagin1998** [comment]
Yeah,  `--augmentate`.

You don't really need `--extraprocesses` (in the end of my `README.md` you can find next words:"Keras has some problems with parallel data augmentation: it creates too many processes. I hope it will be fixed soon, but now it is better to use zero value of --extraprocesses flag (default value);"

So for data augmentation just use `--augmentate` flag.

**Comment by masyagin1998** [comment]
Non-default `--extraprocesses` should make augmentation faster in theory.
But in reality, extra processes for augmentation don't encrease speed too much (but give many problems).

**Comment by LaiSongxuan** [comment]
Thank you for your reply. What's the purpose and effect of the inversion operation (it seems odd to me)? Is it to do with the Dice loss? Because the loss function does not seem to suppress the backgrounds.

**Comment by masyagin1998** [comment]
It is done with 50% possibility for every image. Firstly, it can invert training input image colors with 0.5 possibility, and then take little part of the same image and invert it again with also 0.5 possibility (it inverts only inpit image, not marked image (!!!)).

I used it to train NN to find white text on dark background (LRDE example in README.md)

**Comment by LaiSongxuan** [comment]
Another question. You used Adam with lr=1e-4 for the whole training stage (you mentioned you trained on DIBCO and borders data for 256 epochs)?

**Comment by masyagin1998** [comment]
Yes, I didn't change lr during training.

**Comment by LaiSongxuan** [comment]
Thank you for your information :)

**Comment by mrocr** [comment]
@LaiSongxuan actually the  inversion operation is very smart, it's a good thing.

**Comment by masyagin1998** [comment]
> It is done with 50% possibility for every image. Firstly, it can invert training input image colors with 0.5 possibility, and then take little part of the same image and invert it again with also 0.5 possibility (it inverts only inpit image, not marked image (!!!)).
> 
> I used it to train NN to find white text on dark background (LRDE example in README.md)

50% possibility on first inversion (full inversion) and 25% possibility on second (part inversion), but it doesn't change logic.

**Comment by masyagin1998** [comment]
`        invert = InvertPartAugmentor(0.25)
        p.add_operation(invert)
        p.invert(0.5)
        imgs_in = self.__apply_augmentation__(p)
imgs_in = [p[0] for p in imgs_in]`
